dotnet/Templates/Microsoft.iOS.Templates/ios/Main.cs

```diff
@@ -1,5 +1,12 @@
 using iOSApp1;
 
+#if DEBUG
```

Comment:
I think we can skip updating templates if:

* We have a private _check_for_... field we set when we generate our unmanaged main function (or somehow set this value during build time) - and set it to `true` for debug builds and `false` otherwise.
* In the linker detect if the public CheckFor... setter has been preserved. If not, we can treat the getter as a constant value (because we'll know what we set the private field to).

Reply:
Yes, removing the fields (or properties) is easy (many ways) at least if they are not set from the `.cctor`. But the savings(of two fields) are very minimal since the rest of the logic is already removed. It's mostly a nice side effect :-)

However this keeps the features (and logic) into the magic/unseen land, which is bad for feature discovery. Even when documented a feature is only useful if you know about it (and for .net it won't be the old `mtouch` additional arguments to override them). Having the templates makes them obvious, at least more than finding the new (was it added) correct msbuild syntax.